scripts/prepare_motus_crates/motus_ro_crates_preparer.py

```diff
@@ -220,17 +226,45 @@ def zip_ro_crate_output_folder(self):
     def clean_up(self):
         shutil.rmtree(self.downloaded_ro_crate_zip_temp_dir)
 
+    def extract_files_from_subdirectories(self, url):
+        response = requests.get(url)
+        if response.status_code != 200:
+            print(f"Failed to fetch {url}")
+            return
+        soup = BeautifulSoup(response.text, 'html.parser')
+        links = soup.find_all('a')
```

Comment:
Nice approach with `bs4`! However, given that this is also a real FTP server (not just a HTTP rendering of one), it might be a bit lighter / more explicit to do this using `ftplib` from the standard library (thereby dropping a fairly heavy dependency too):
```python
from ftplib import FTP
ftp = FTP('ftp.ebi.ac.uk')
ftp.login()
ftp.cwd('pub/databases/metagenomics/motus_pipeline/')  # or from the `original_ro_crate_path` I guess
ftp.nlst()
#outputs: ['ERP001736', 'ERP010880', 'ERP156184', 'SRP039390', 'SRP109831', 'SRP110813', 'SRP278138', 'motus_tables', 'run.sh']
```
and I think you can do a similar recursive search approach like that: you can check whether an item returned by `nlst` is itself a directory using `is_dir = ftp.nlst(item) == []`.

Reply:
After discussions with @SandyRogers we agreed to proceed with the BS4 approach